scripts/upgrade.sh

```diff
@@ -14,7 +14,26 @@
 # limitations under the License.
 #
 # (c) 2015, Nolan Brubaker <nolan.brubaker@rackspace.com>
-set -eux pipefail
+set -ux pipefail
+
+FAILED=0
+
+function run_or_print() {
+        command="$@"
+        if [ $FAILED -ne 0 ]; then
+                echo "  ${command}"
+        else
+                eval "$command"
+                FAILED=$?
+                if [ $FAILED -ne 0 ]; then
+                    echo "******************** FAILURE ********************"
+                    echo "The upgrade script has failed. Please rerun the following task to continue"
+                    echo "Failed on task ${command}"
+                    echo "Do NOT rerun the upgrade script!"
+                    echo "Please execute the remaining tasks:"
```

Comment:
As we discussed in the channel, we actually are printing the remaining tasks so that the user won'y have to look at the script's contents.